### PR TITLE
resolves #45 that the validation works first time when rendered

### LINKED ISSUE #45: validation not appearing by default
when validation should be appearing during the first rendering. instead it will only appear during the save button triggering.

exploration has found that there is also the following dials in the useform code:

onChange | string | Validation is triggered on the changeevent for each input, leading to multiple re-renders. Warning: this often comes with a significant impact on performance.
-- | -- | --
onTouched | string | Validation is initially triggered on the first blur event. After that, it is triggered on every change event.

but these only work on the given field as its touched.

I would think theres some way to make it trigger by default. but trying form.validate() causes a infinite loop. more exploration needed.

### PR DESCRIPTION
**Pull request by alan-stokes**
resolves issue #45 on the new proposal page and new observation that the validation is displayed by default when making a new observation / proposal / investigators / . 

have done so by utilising the useEffect hook. works locally. 

have avoided targets due to its major rejig on https://github.com/orppst/pst-gui/pull/40
once thats in, i can revisit this to add the hook in the right file.

### CODE REVIEW

**Comment by alan-stokes** [comment]
updated with master nov 13th 9.17am

**Comment by DJWalker42** [comment]
I'd like other peoples opinion on this design choice please

File: src/main/webui/src/technicalGoals/edit.group.tsx [hunk 1]
```diff
@@ -53,8 +76,8 @@ export default function TechnicalGoalEditGroup(props: TechnicalGoalProps ):
 
     // integers specifying the proportional number of columns for the performance parameter
     // section and the spectral window section
-    const TOTAL_COLUMNS = 10;
```
Comment (alan-stokes) [inline src/main/webui/src/technicalGoals/edit.group.tsx:56]: oops. this was me trying to make the validation message stay on one line. It wasnt intentional to change your layout. Please feel free to demand that change back if you want.

Comment (alan-stokes) [reply src/main/webui/src/technicalGoals/edit.group.tsx:56]: it was me messing with for this discussion as well: https://github.com/orgs/orppst/discussions/13

**Comment by AllanEngland** [comment]
Yeah, for me this goes back to the conflation of errors and requirements that we've seen on the ESO demo site.

Personally, I'm all for highlighting what needs to be done to complete a form beyond simple asterisks, but I'm not a big fan of painting it all red in the same way as an error.

Is there a practical two stage method so an initial form load has informative directional prompts that can become errors when saved/submitted?   If not then I feel we should work on the copy to be more polite and fit the messages to be informative and instructional (if that makes sense to anyone other than me!)

**Comment by alan-stokes** [comment]
thinking it through as ideas. 

1. if you really don't like the validation being shown right away. I guess you could add placeholders inside the boxes. so that they are not red. The issue i see with that is that its not immediately apparent what the user needs to do.  There's no "hard of sight" peeps being focused on the red.
2. You could have the 1/2 way house with acceptable defaults, but that's had its issues on the same boat of "how does the user know what to do". and has tie ins to automatic savings.
3. if we want a valid form before saving, we also need to disable the form save button till valid and whilst pristine, which means the first moment a user gets to save, it'll go "all red" anyhow, and give the user this "why didn't you tell me this first" kinda experience. This is what I'd call a ALL mode.
3.2 if we have auto save, without default values, we've got more work to ensure invalid data can be saved. such as saving text in a number field (assuming we didn't have that locked down in other ways etc). And this version asks the question, what is the use of validation then. This id call the NOTHING mode.

at moment we seem to be in none of these zones. as i can hit save right away, and be pushed back with the spectral point field. but not about the other fields in performance params.  It feels like that field has higher authority than the rest of the form, which seems odd to me (given the mental state of wanting either 2 or 3)

any other views?

**Comment by alan-stokes** [comment]
by the way, whats the ESO demo site?

**Comment by AllanEngland** [comment]
The ESO public p1 demo is here: https://www.eso.org/p1demo/proposals

**Comment by alan-stokes** [comment]
as an aside. this is a UI bezt practices that I kinda agree with. just for food for thought:
https://designlab.com/blog/form-ui-design-best-practices/#:~:text=Clearly%20mark%20optional%20fields&text=While%20marking%20optional%20fields%20with,with%20the%20text%20(optional).

**Comment by alan-stokes** [comment]
no decision has been made in relation to this pr. so leaving, as having validation running by defacto is still a open question, and the work in here can be repurporsed if it was deemed the way to go.

**Comment by DJWalker42** [comment]
I've closed #45 as obsolete so this PR can also be closed

**Pull request closed without merging**